File: Java/CesiumLanguageWriterTests/translatedSrc/cesiumlanguagewritertests/TestPixelOffsetCesiumWriter.java
package cesiumlanguagewritertests;


import agi.foundation.compatibility.*;
import agi.foundation.compatibility.MapHelper;
import agi.foundation.compatibility.TestContextRule;
import agi.foundation.compatibility.Using;
import cesiumlanguagewriter.*;
import cesiumlanguagewriter.advanced.*;
import java.util.Map;
import javax.annotation.Nonnull;
import org.junit.ClassRule;
import org.junit.FixMethodOrder;
import org.junit.Rule;
import org.junit.runners.MethodSorters;
import org.junit.Test;

@SuppressWarnings({
    "unused",
    "deprecation",
    "serial"
})
@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class TestPixelOffsetCesiumWriter extends TestCesiumInterpolatablePropertyWriter<PixelOffsetCesiumWriter> {
    @Test
    public final void writeCartesian2() {
        Rectangular expected = new Rectangular(10.0, 20.0);
        try (Using<PacketCesiumWriter> using$0 = new Using<PacketCesiumWriter>(openPacket())) {
            final PacketCesiumWriter packet = using$0.resource;
            try (Using<BillboardCesiumWriter> using$1 = new Using<BillboardCesiumWriter>(packet.openBillboardProperty())) {
                final BillboardCesiumWriter billboard = using$1.resource;
                try (Using<BillboardCesiumWriter> using$2 = new Using<BillboardCesiumWriter>(billboard.openInterval())) {
                    final BillboardCesiumWriter interval = using$2.resource;
                    try (Using<PixelOffsetCesiumWriter> using$3 = new Using<PixelOffsetCesiumWriter>(interval.openPixelOffsetProperty())) {
                        final PixelOffsetCesiumWriter pixelOffset = using$3.resource;
                        pixelOffset.writeCartesian2(expected);
                    }
                }
            }
        }
        final Map<String, Object> tempCollection$0 = MapHelper.create();
        MapHelper.add(tempCollection$0, BillboardCesiumWriter.PixelOffsetPropertyName, expected);
        assertExpectedJson(PacketCesiumWriter.BillboardPropertyName, tempCollection$0);
    }

    @Test
    public final void writeReference() {
        final String expectedIdentifier = "someId";
        final String expectedPropertyName = "pixelOffset";
        try (Using<PacketCesiumWriter> using$0 = new Using<PacketCesiumWriter>(openPacket())) {
            final PacketCesiumWriter packet = using$0.resource;
            try (Using<BillboardCesiumWriter> using$1 = new Using<BillboardCesiumWriter>(packet.openBillboardProperty())) {
                final BillboardCesiumWriter billboard = using$1.resource;
                try (Using<BillboardCesiumWriter> using$2 = new Using<BillboardCesiumWriter>(billboard.openInterval())) {
                    final BillboardCesiumWriter interval = using$2.resource;
                    try (Using<PixelOffsetCesiumWriter> using$3 = new Using<PixelOffsetCesiumWriter>(interval.openPixelOffsetProperty())) {
                        final PixelOffsetCesiumWriter pixelOffset = using$3.resource;
                        pixelOffset.writeReference(expectedIdentifier, expectedPropertyName);
                    }
                }
            }
        }
        final Map<String, Object> tempCollection$1 = MapHelper.create();
        MapHelper.add(tempCollection$1, PixelOffsetCesiumWriter.ReferencePropertyName, expectedIdentifier + "#" + expectedPropertyName);
        final Map<String, Object> tempCollection$0 = MapHelper.create();
        MapHelper.add(tempCollection$0, BillboardCesiumWriter.PixelOffsetPropertyName, tempCollection$1);
        assertExpectedJson(PacketCesiumWriter.BillboardPropertyName, tempCollection$0);
    }

    @Test
    public final void testDeleteProperty() {
        final boolean expectedDelete = true;
        try (Using<PacketCesiumWriter> using$0 = new Using<PacketCesiumWriter>(openPacket())) {
            final PacketCesiumWriter packet = using$0.resource;
            try (Using<BillboardCesiumWriter> using$1 = new Using<BillboardCesiumWriter>(packet.openBillboardProperty())) {
                final BillboardCesiumWriter billboard = using$1.resource;
                try (Using<BillboardCesiumWriter> using$2 = new Using<BillboardCesiumWriter>(billboard.openInterval())) {
                    final BillboardCesiumWriter interval = using$2.resource;
                    try (Using<PixelOffsetCesiumWriter> using$3 = new Using<PixelOffsetCesiumWriter>(interval.openPixelOffsetProperty())) {
                        final PixelOffsetCesiumWriter pixelOffset = using$3.resource;
                        pixelOffset.writeDelete(expectedDelete);
                    }
                }
            }
        }
        final Map<String, Object> tempCollection$1 = MapHelper.create();
        MapHelper.add(tempCollection$1, PixelOffsetCesiumWriter.DeletePropertyName, expectedDelete);
        final Map<String, Object> tempCollection$0 = MapHelper.create();
        MapHelper.add(tempCollection$0, BillboardCesiumWriter.PixelOffsetPropertyName, tempCollection$1);
        assertExpectedJson(PacketCesiumWriter.BillboardPropertyName, tempCollection$0);
    }

    @Override
    protected CesiumPropertyWriter<PixelOffsetCesiumWriter> createPropertyWriter(String propertyName) {
        return new PixelOffsetCesiumWriter(propertyName);
    }

    @Nonnull
    private static final TestContextRule rule$testContext = new TestContextRule();

    @Nonnull
    @Rule
    @ClassRule
    public static TestContextRule getRule$testContext() {
        return rule$testContext;
    }
}

File: DotNet/CesiumLanguageWriterTests/TestPixelOffsetCesiumWriter.cs
using System.Collections.Generic;
using CesiumLanguageWriter;
using CesiumLanguageWriter.Advanced;
using JetBrains.Annotations;
using NUnit.Framework;

namespace CesiumLanguageWriterTests
{
    [TestFixture]
    public class TestPixelOffsetCesiumWriter : TestCesiumInterpolatablePropertyWriter<PixelOffsetCesiumWriter>
    {
        [Test]
        public void WriteCartesian2()
        {
            var expected = new Rectangular(10.0, 20.0);

            using (var packet = OpenPacket())
            using (var billboard = packet.OpenBillboardProperty())
            using (var interval = billboard.OpenInterval())
            using (var pixelOffset = interval.OpenPixelOffsetProperty())
            {
                pixelOffset.WriteCartesian2(expected);
            }

            AssertPropertyJson(BillboardCesiumWriter.PixelOffsetPropertyName, expected);
        }

        [Test]
        public void WriteReference()
        {
            const string expectedIdentifier = "someId";
            const string expectedPropertyName = "pixelOffset";

            using (var packet = OpenPacket())
            using (var billboard = packet.OpenBillboardProperty())
            using (var interval = billboard.OpenInterval())
            using (var pixelOffset = interval.OpenPixelOffsetProperty())
            {
                pixelOffset.WriteReference(expectedIdentifier, expectedPropertyName);
            }

            AssertPropertyJson(BillboardCesiumWriter.PixelOffsetPropertyName, new Dictionary<string, object>
            {
                { PixelOffsetCesiumWriter.ReferencePropertyName, expectedIdentifier + "#" + expectedPropertyName },
            });
        }

        [Test]
        public void TestDeleteProperty()
        {
            const bool expectedDelete = true;

            using (var packet = OpenPacket())
            using (var billboard = packet.OpenBillboardProperty())
            using (var interval = billboard.OpenInterval())
            using (var pixelOffset = interval.OpenPixelOffsetProperty())
            {
                pixelOffset.WriteDelete(expectedDelete);
            }

            AssertPropertyJson(BillboardCesiumWriter.PixelOffsetPropertyName, new Dictionary<string, object>
            {
                { PixelOffsetCesiumWriter.DeletePropertyName, expectedDelete },
            });
        }

        private void AssertPropertyJson([NotNull] string propertyName, [NotNull] object value)
        {
            AssertExpectedJson(PacketCesiumWriter.BillboardPropertyName, new Dictionary<string, object>
            {
                { propertyName, value },
            });
        }

        protected override CesiumPropertyWriter<PixelOffsetCesiumWriter> CreatePropertyWriter(string propertyName)
        {
            return new PixelOffsetCesiumWriter(propertyName);
        }
    }
}
